test/WebJobs.Script.Tests.Integration/ServiceBus/ServiceBusEndToEndTestBase.cs

```diff
@@ -0,0 +1,144 @@
+using Microsoft.Azure.ServiceBus.Core;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.WindowsAzure.Storage.Blob;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Script.Tests.Integration.ServiceBus
+{
+    public abstract class ServiceBusEndToEndTestsBase<TTestFixture> :
+        EndToEndTestsBase<TTestFixture> where TTestFixture : ServiceBusTestFixture, new()
+    {
+        public ServiceBusEndToEndTestsBase(TTestFixture fixture) : base(fixture)
+        {
+        }
+
+        protected async Task ServiceBusQueueTriggerAndOutputTest()
+        {
+            // ServiceBus tests need the following environment vars:
+            // "AzureWebJobsServiceBus" -- the connection string to the account
+
+            var resultBlob = Fixture.TestOutputContainer.GetBlockBlobReference("servicebuse2e-queue-completed");
+            await resultBlob.DeleteIfExistsAsync();
+
+            await Fixture.CleanUpEntity(Fixture.TestQueueName1);
+            await Fixture.CleanUpEntity(Fixture.TestQueueName2);
+            // trigger
+            await Fixture.SendQueueMessage(Fixture.TestQueueName1, "test-queue");
+
+            // now wait for function to be invoked
+            string result = await WaitForBlobWithTimeout(resultBlob);
+
+            var list = await Fixture.CleanUpEntity(Fixture.TestQueueName2);
+            Assert.Equal(list.Count, 1);
+            Assert.Equal(list[0], "test-queue-completed");
+        }
+
+        protected async Task ServiceBusTopicTriggerTest()
+        {
+            var resultBlob = Fixture.TestOutputContainer.GetBlockBlobReference("servicebuse2e-topic-completed");
+            await resultBlob.DeleteIfExistsAsync();
+            
+            await Fixture.CleanUpEntity(Fixture.TestTopicName1);
+            // trigger
+            await Fixture.SendTopicMessage(Fixture.TestTopicName1, "test-topic");
+
+            // now wait for function to be invoked
+            string result = await WaitForBlobWithTimeout(resultBlob);
+
+            Assert.Equal(result, "test-topic-completed");
+        }
+
+        protected async Task ServiceBusTopicOutputTest()
+        {
+
+            var resultBlob = Fixture.TestOutputContainer.GetBlockBlobReference("servicebuse2e-topic-completed");
+            await resultBlob.DeleteIfExistsAsync();
+
+            await Fixture.CleanUpEntity(Fixture.TestTopicName2);
+            await Fixture.Host.BeginFunctionAsync("ServiceBusTopicOutput", "test-topic");
+
+            // now wait for function to be invoked
+            string result = await WaitForBlobWithTimeout(resultBlob);
+
+            var list = await Fixture.CleanUpEntity(Fixture.TestTopicName2);
+            Assert.Equal(list.Count, 1);
+            Assert.Equal(list[0], "test-topic-completed");
+        }
+
+        private async Task<string> WaitForBlobWithTimeout(CloudBlockBlob resultBlob)
```

Comment:
You can get rid of this -- the WaitForBlobAndGetStringAsync() will timeout already. It ultimately calls `TestHelpers.Await()`, which has settings for `timeout` and `throwWhenDebugging`, which you can adjust. I'm guessing you were debugging and not ever seeing the timeout occur? If so, that's b/c `throwWhenDebugging` defaults to `false`. If you want more control over that, you can expose those parameters on `WaitForBlobAndGetStringAsync` rather than using another timeout mechanism.